packages/mantine/src/components/table/layouts/RowLayout.tsx

```diff
@@ -0,0 +1,150 @@
+
+interface TableStylesParams {
+    multiRowSelectionEnabled: boolean;
+    disableRowSelection: boolean;
+}
+
+const useStyles = createStyles<string, TableStylesParams>((theme, {multiRowSelectionEnabled, disableRowSelection}) => {
+    const rowBackgroundColor =
+        theme.colorScheme === 'dark'
+            ? theme.fn.rgba(theme.colors[theme.primaryColor][7], 0.2)
+            : theme.colors[theme.primaryColor][0];
+    return {
+        headerColumns: {
+            '& th:first-of-type > *': {
+                paddingLeft: theme.spacing.xl,
+            },
+
+            '& input[type=checkbox]': {
+                backgroundColor: disableRowSelection ? `${theme.colors.gray[2]}` : undefined,
+                borderColor: disableRowSelection ? `${theme.colors.gray[3]}` : `${theme.colors.gray[4]}`,
+                pointerEvents: disableRowSelection ? 'none' : 'auto',
+                cursor: disableRowSelection ? 'not-allowed' : 'default',
+
+                '& + svg': {
+                    color: disableRowSelection ? `${theme.colors.gray[5]}` : 'inherit',
+                },
+            },
+        },
+
+        rowSelected: {
+            backgroundColor: multiRowSelectionEnabled ? undefined : rowBackgroundColor,
+        },
+
+        rowUnselectable: {
+            '& input[type=checkbox]': {
+                backgroundColor: `${theme.colors.gray[2]}`,
+                borderColor: `${theme.colors.gray[3]}`,
+                pointerEvents: 'none',
+                cursor: 'not-allowed',
+
+                '&:checked + svg': {
+                    color: `${theme.colors.gray[5]}`,
+                },
+            },
+        },
+
+        rowCollapsibleButtonCell: {
+            textAlign: 'right',
+            padding: `calc(${theme.spacing.xs}/2) ${theme.spacing.sm} !important`,
+        },
+
+        row: {
+            '&:hover': {
+                backgroundColor: rowBackgroundColor,
+            },
+        },
+    };
+});
+
+const RowLayoutHeader = <T,>({table}: TableLayoutProps<T>) => {
+    const {multiRowSelectionEnabled, disableRowSelection} = useTable();
+    const {classes} = useStyles({disableRowSelection, multiRowSelectionEnabled});
+    const headers = table.getHeaderGroups().map((headerGroup) => (
+        <tr key={headerGroup.id} className={classes.headerColumns}>
+            {headerGroup.headers.map((columnHeader) => (
+                <Th key={columnHeader.id} header={columnHeader} />
+            ))}
+        </tr>
+    ));
+    return <>{headers}</>;
+};
+
+const RowLayoutBody = <T,>({table, doubleClickAction, getExpandChildren, loading}: TableLayoutProps<T>) => {
+    const {multiRowSelectionEnabled, disableRowSelection} = useTable();
+    const {classes, cx} = useStyles({disableRowSelection, multiRowSelectionEnabled});
+
+    const rows = table.getRowModel().rows.map((row) => {
+        const rowChildren = getExpandChildren?.(row.original) ?? null;
+        const isSelected = !!row.getIsSelected();
```

Comment:
out of curiosity. This function does not return a boolean ?

Reply:
It does, good catch. I mainly just copy pasted what we already had in the table component btw.